### PR TITLE
Enhancements to load balancer CRUD tests and fixes in discovered LB impl issues

### PR DESCRIPTION
**Pull request by martinsawicki**

### CODE REVIEW

File: azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/LoadBalancerImpl.java [hunk 1]
```diff
@@ -171,22 +184,29 @@ protected void beforeCreating()  {
...
-                        .withExistingLoadBalancerBackend(this, backendName)
-                        .parent()
+                    .withExistingLoadBalancerBackend(this, backendName)
+                    .parent()
                     .apply();
             } catch (Exception e) {
-                e.printStackTrace();
+                nicExceptions.add(e);
             }
         }
```
Comment (anuchandy) [inline azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/LoadBalancerImpl.java]: Here we try `[1] fetch` and `[2] update` a list of NICs i.e. `processing` each NIC involves two API calls. We aggregate any exception happens during `processing` of each NIC and finally throw it.  In this case, shouldn't we include `[1] fetch` also in the `try` block ?

Comment (anuchandy) [inline azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/LoadBalancerImpl.java]: I mean this:

We `[1] fetch` and `[2] update` a list of NICs i.e. `processing` each NIC involves two API calls. But we aggregate only  the exception happens during `[2] update` of each NIC and finally throw it. In this case, shouldn't we include `[1] fetch` also in the try block ?

Comment (martinsawicki) [inline azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/LoadBalancerImpl.java]: good point; changed in the newest commit

**Comment by anuchandy** [comment]
lgtm

**Pull request merged**